llarp/util/time.hpp

```diff
@@ -29,7 +25,64 @@ namespace llarp
   struct time_delta
   {
     const TimePoint_t at;
+
+    /// how far away are we from the time point we have
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    delta() const
+    {
+      return distance_from<delta_res_t>(DateClock_t::now());
+    }
+
+    template <typename delta_res_t = std::chrono::milliseconds>
+    auto
+    distance_from(const TimePoint_t& other) const
+    {
+      return std::chrono::duration_cast<delta_res_t>(other - at);
+    }
   };
+
+  /// get a time delta with a timepoint in the future
+  template <typename Time_Duration>
+  TimePoint_t
+  future_time_point(Time_Duration future)
+  {
+    return DateClock_t::now() + future;
+  }
+
+  /// get a time delta with a timepoint in the future
+  template <typename Time_Duration>
+  auto
+  future_time_delta(Time_Duration future)
+  {
+    return time_delta<Time_Duration>{future_time_point(future)};
+  }
+
+  /// returns the time now as a time point
+  inline TimePoint_t
+  time_now()
+  {
+    return DateClock_t::now();
+  }
+
+  /// convert a timepoint to a unix millis stamp
+  uint64_t
+  to_unix_stamp(const TimePoint_t& t);
+
+  /// get time right now as milliseconds, this is monotonic
+  inline Duration_t
+  time_now_ms()
+  {
+    return Duration_t{to_unix_stamp(time_now())};
+  }
+
+  /// convert a legacy time stamp to a time point
+  template <typename Time_Duration = std::chrono::milliseconds>
+  auto
+  to_time_point(llarp_time_t stamp)
+  {
+    return future_time_point<Time_Duration>(stamp - time_now_ms());
```

Comment:
I think this is the same as: `return DateClock_t{stamp};`

Reply:
`DateClock_t` is not a time point type.